### PR TITLE
Weapon/AI/Detection/FAR HPcalc tweaks

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
- Fix a number of issues with Pwing+FAR HP calc.
	- PWing Thickness-based Mass+HP now takes PWing Edges into account if PWing Edge Lift enabled.
	- Fix armored cockpit mass resetting on load.
	- Add search tags to all BDA parts.
	- Fix AI potentially seeing beyond viewrange.
	- Increase preemptive lead distance AI will begin leading target before entering engagement range.
	- AI will now switch off of Afterburner if AB is out of fuel (for Rapiers/other LF/non-air AB-mode engines).
	- Adds optional tracer smoke.
		- new 'smokeTexturePath' setting for ModuleWeapon to set custom smoke trail FX, leave blank for default bullet FX
	- Properly support weapons with multiple ModuleWeapon modules.
	- Add turret support for Multimissile Launchers.
	-Turret integration with ModuleMultiLauncher for MML style Hydra Turrets, etc.
       - Add rcsOverride field to ModuleECMJammer to enable setting craft RCS to a set value.
       - Split CM time into missile evade time and CM deploy time to allow setting a craft to begin evading, then deploy CM, etc
       - RWRs now activate when they first detect a missile instead of when missile enters visual range.
	- RWRs now track active Radar missiles regardless of missile visibility if VARIABLE_MISSILE_VISIBILITY is enabled.

### CODE REVIEW

File: BDArmory/Distribution/GameData/BDArmory/Parts/ArmorPlate/IsoTripanelArmor.cfg [hunk 1]
```diff
@@ -26,7 +26,7 @@ title = #loc_BDArmory_part_TriIsoPanel_title //BD Armor Panel Oblique Triangle
 manufacturer = #loc_BDArmory_agent_title //Bahamuto Dynamics
 description = #loc_BDArmory_part_Tripanel_description //A sturdy Universal Structural Panel that can be configured to be a variety of sizes and use a variety of materials, perfect for constructing or armoring all sorts of things.
 attachRules = 1,1,1,1,1
-
+tags = #loc_BDArmory_part_bahaArmor_tags
```
Comment (BrettRyland) [inline BDArmory/Distribution/GameData/BDArmory/Parts/ArmorPlate/IsoTripanelArmor.cfg:29]: Several parts have multiple `tags` entries from these changes:
```
find Distribution/GameData/BDArmory/Parts/ -name '*.cfg' -exec grep -crI 'tags =' {} + |grep -v ':0$\|:1$'|sort
Distribution/GameData/BDArmory/Parts/ArmorPlate/IsoTripanelArmor.cfg:2
Distribution/GameData/BDArmory/Parts/ArmorPlate/panelArmor.cfg:2
Distribution/GameData/BDArmory/Parts/ArmorPlate/TripanelArmor.cfg:2
Distribution/GameData/BDArmory/Parts/EJ200/caesar_hone_60.cfg:2
Distribution/GameData/BDArmory/Parts/Mk1OpenCockpit/mk1opencockpit_RP_Type2.cfg:2
Distribution/GameData/BDArmory/Parts/Mk1OpenCockpit/weaponizedcommandseat.cfg:2
Distribution/GameData/BDArmory/Parts/ReactiveArmor/ReactiveArmor.cfg:2
Distribution/GameData/BDArmory/Parts/SaturnAL31/SaturnAL31.cfg:2
Distribution/GameData/BDArmory/Parts/StingRayBDA/StingRayBDATorpedo.cfg:2
```

File: BDArmory/Control/MissileFire.cs [hunk 2]
```diff
@@ -1939,6 +1943,21 @@ IEnumerator GuardMissileRoutine()
                                     yield return new WaitForFixedUpdate();
                                 }
                             }
+                            if (guardTarget && ml && mlauncher.multiLauncher.turret && vesselRadarData.locked)
```
Comment (BrettRyland) [inline BDArmory/Control/MissileFire.cs:1946]: This causes an NRE if `mlauncher.multiLauncher` is `null`.
Also in other places where `.multiLauncher` is accessed: lines 2015, 2076, 2140, 2196, 2226, 2310, 6463.

File: BDArmory/Distribution/GameData/BDArmory/Parts/EJ200/caesar_hone_60.cfg [hunk 3]
```diff
@@ -26,6 +26,7 @@ PART
 	manufacturer = Twin Crown Aerospace Industries
 	description = Word is that this engine was the result of international cooperation, which produced an engine with exceptional performance and potential.
 	attachRules = 1,0,1,0,0
+	tags = bda after aircraft burner engine fighter jet (typhoon plane propuls
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Distribution/GameData/BDArmory/Parts/EJ200/caesar_hone_60.cfg:29]: Tag list was copied from the stock Panther. As for the parenthesis, its included in stock tag lists (generally before the part name), though I couldn't tell you why.

[the remark this answers is not in the document]
Comment (BrettRyland) [reply BDArmory/Distribution/GameData/BDArmory/Parts/EJ200/caesar_hone_60.cfg:29]: Huh, weird. OK.

**Comment by BrettRyland** [comment]
I'm still setting up my new computer so it might take a day or two before I can merge this.

**Pull request merged**